File: maplibre_ios/ios/maplibre_ios/Sources/maplibre_ios/MapLibreRegistry.m
#import <Foundation/Foundation.h>
#import "MapLibreRegistry.h"

@implementation MapLibreRegistry

static NSMutableDictionary<NSNumber *, MLNMapView *> *mapRegistry;
static id _activity;
static id _context;

// Initialize the static map registry
+ (void)initialize {
    if (self == [MapLibreRegistry class]) {
        mapRegistry = [NSMutableDictionary dictionary];
    }
}

// Method to get the map for a given viewId
+ (nullable MLNMapView *)getMapWithViewId:(int64_t)viewId {
    return mapRegistry[@(viewId)];
}

// Method to add a map to the registry
+ (void)addMapWithViewId:(int64_t)viewId map:(MLNMapView *)map {
    mapRegistry[@(viewId)] = map;
}

// Getter and Setter for activity
+ (id)activity {
    return _activity;
}

+ (void)setActivity:(id)activity {
    _activity = activity;
}

// Getter and Setter for context
+ (id)context {
    return _context;
}

+ (void)setContext:(id)context {
    _context = context;
}

@end

File: packages/maplibre_ios/ios/maplibre_ios/Sources/maplibre_ios/MapLibreRegistry.swift
import Foundation
import MapLibre
import UIKit

// Update the header file for this class like this:
// cd maplibre_ios/ios/maplibre_ios/Sources/maplibre_ios/
// ./gen_swift_headers.sh

@objc(MapLibreRegistry)
public class MapLibreRegistry: NSObject {
    private static var mapRegistry: [Int64: MLNMapView] = [:]

    /// Method to get the map for a given viewId
    @objc public static func getMap(viewId: Int64) -> MLNMapView? {
        mapRegistry[viewId]
    }

    /// Method to add a map to the registry
    public static func addMap(viewId: Int64, map: MLNMapView) {
        mapRegistry[viewId] = map
    }

    /// Method to remove a map to the registry
    public static func removeMap(viewId: Int64) {
        mapRegistry.removeValue(forKey: viewId)
    }

    /// Warning: Storing Activity in a static field may lead to memory leaks.
    @objc public static var activity: AnyObject?

    /// Warning: Storing Context in a static field may lead to memory leaks.
    @objc public static var context: AnyObject?
}
